Lib/tkinter/auto_complete.py

```diff
@@ -0,0 +1,155 @@
+    Methods:
+    __init__ -- Set up the `tk.Listbox` and `tk.Entry` widgets
+    build -- Build a list of autocompletion entries
+    _update_autocomplete -- Internal method
+    _select_entry -- Internal method
+    _cycle_up -- Internal method
+    _cycle_down -- Internal method
+    Other attributes:
+    text -- StringVar object associated with the `tk.Entry` widget
+    entry -- The `tk.Entry` widget (access this directly if you
+             need to change styling)
+    listbox -- The `tk.Listbox` widget (access this directly if
+             you need to change styling)
+    """
+    LISTBOX_HEIGHT = 5
+    LISTBOX_WIDTH = 25
+    ENTRY_WIDTH = 25
+
+    def __init__(self, master, *args, **kwargs):
+        """Constructor.
+        Create the `self.entry` and `self.listbox` widgets.
+        Note that these widgets are not yet displayed and will only
+        be visible when you call `self.build`.
+        Arguments:
+        master -- The master tkinter widget
+        Returns:
+        None
+        """
+        super(AutocompleteEntry, self).__init__(*args, **kwargs)
+        self.text = tk.StringVar()
+        self.entry = tk.Entry(
+            self,
+            textvariable=self.text,
+            width=self.ENTRY_WIDTH
+        )
+        self.listbox = tk.Listbox(
+            self,
+            height=self.LISTBOX_HEIGHT,
+            width=self.LISTBOX_WIDTH
+        )
+
+    def build(
+              self,
+              entries,
+              max_entries=5,
+              case_sensitive=False,
+              no_results_message=NO_RESULTS_MESSAGE
+            ):
+        """Set up the autocompletion settings.
+        Binds <KeyRelease>, <<ListboxSelect>>, <Down> and <Up> for
+        smooth cycling between autocompletion entries.
+        Arguments:
+        entries -- An iterable containg autocompletion entries (strings)
+        max_entries -- [int] The maximum number of entries to display
+        case_sensitive -- [bool] Set to `True` to make autocompletion
+                          case-sensitive
+        no_results_message -- [str] Message to display when no entries
+                              match the current entry; you can use a
+                              formatting identifier '{}' which will be
+                              replaced with the entry at runtime
+        Returns:
+        None
+        """
+        if not case_sensitive:
+            entries = [entry.lower() for entry in entries]
+
+        self._case_sensitive = case_sensitive
+        self._entries = entries
+        self._no_results_message = no_results_message
+        self._listbox_height = max_entries
+
+        self.entry.bind("<KeyRelease>", self._update_autocomplete)
+        self.entry.focus()
+        self.entry.grid(column=0, row=0)
+
+        self.listbox.bind("<<ListboxSelect>>", self._select_entry)
+        self.listbox.grid(column=0, row=1)
+        self.listbox.grid_forget()
```

Comment:
```suggestion
        self.listbox.grid_remove()
```
This way, the previous grid options will be remembered.